src/VisualStudio/Core/Impl/CodeModel/CodeModelIncrementalAnalyzer.cs

```diff
@@ -70,31 +70,36 @@ public void RemoveDocument(DocumentId documentId)
 
             public void FireEvents(DocumentId documentId, CancellationToken cancellationToken)
             {
-                var project = _workspace.DeferredState.ProjectTracker.GetProject(documentId.ProjectId);
-                if (project == null)
+                _notificationService.RegisterNotification(() =>
                 {
-                    return;
-                }
-
-                var codeModelProvider = project as IProjectCodeModelProvider;
-                if (codeModelProvider == null)
-                {
-                    return;
-                }
-
-                var filename = _workspace.GetFilePath(documentId);
-                if (filename == null)
-                {
-                    return;
-                }
-
-                if (!codeModelProvider.ProjectCodeModel.TryGetCachedFileCodeModel(filename, out var fileCodeModelHandle))
-                {
-                    return;
-                }
-
-                var codeModel = fileCodeModelHandle.Object;
-                _notificationService.RegisterNotification(() => codeModel.FireEvents(), _listener.BeginAsyncOperation("CodeModelEvent"), cancellationToken);
+                    var project = _workspace.DeferredState.ProjectTracker.GetProject(documentId.ProjectId);
```

Comment:
@jasonmalinowski any concerns here that `DeferredState` wouldn't be created?  I guess it never get's *uncreated*, and this will never happen earlier than before.